### PR TITLE
[ISSUE #2495] Add mongodb connector producer

### LINKED ISSUE #2495: [Feature] Mongodb connector producer
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Feature Request

Add mongodb connector producer

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mroccyen**
Fixes #2495.

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbReplicaSetProducer.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.cloudevents.CloudEvent;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+
+public class MongodbReplicaSetProducer implements Producer {
+    private static final Logger logger = LoggerFactory.getLogger(MongodbReplicaSetProducer.class);
```
Comment (mxsm) [inline eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbReplicaSetProducer.java:42]: How about use lombok annotation @Slf4j

File: eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbStandaloneProducer.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import io.cloudevents.CloudEvent;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+
+@SuppressWarnings("all")
+public class MongodbStandaloneProducer implements Producer {
+
+    private static final Logger logger = LoggerFactory.getLogger(MongodbStandaloneProducer.class);
```
Comment (mxsm) [inline eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbStandaloneProducer.java:47]: How about use lombok annotation @slf4j

File: eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbReplicaSetProducer.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+import org.apache.eventmesh.api.RequestReplyCallback;
+import org.apache.eventmesh.api.SendCallback;
+import org.apache.eventmesh.api.SendResult;
+import org.apache.eventmesh.api.exception.ConnectorRuntimeException;
+import org.apache.eventmesh.api.exception.OnExceptionContext;
+import org.apache.eventmesh.api.producer.Producer;
+import org.apache.eventmesh.connector.mongodb.client.MongodbClientManager;
+import org.apache.eventmesh.connector.mongodb.config.ConfigurationHolder;
+import org.apache.eventmesh.connector.mongodb.utils.MongodbCloudEventUtil;
+
```
Comment (mxsm) [inline eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbReplicaSetProducer.java:29]: MongodbCloudEventUtil class not find

File: eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbReplicaSetProducer.java (hunk 1 above)
Comment (mroccyen) [reply eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbReplicaSetProducer.java:42]: @xwm1992 is there such a requirement for use?  I'll modify it if there is.

Comment (mxsm) [reply eventmesh-connector-plugin/eventmesh-connector-mongodb/src/main/java/org/apache/eventmesh/connector/mongodb/producer/MongodbReplicaSetProducer.java:42]: > @xwm1992 is there such a requirement for use? I'll modify it if there is.

Master branch that I have done this.

**Review by mxsm — APPROVED** [review]
LGTM~

**Comment by mxsm** [comment]
Hi @mroccyen Currently, the project is separating the Connector and Storage components. You can wait to see the module format of Storage and then modify this portion to become part of the Storage module. This may require modifying the module and package names.  @xwm1992

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**